pkg/storage/postgres/postgres.go

```diff
@@ -612,7 +612,7 @@ func (p *Postgres) ReadChanges(
 		From("changelog").
 		Where(sq.Eq{"store": store}).
 		Where(fmt.Sprintf("inserted_at < NOW() - interval '%dms'", horizonOffset.Milliseconds())).
-		OrderBy("inserted_at asc")
+		OrderBy("ulid asc")
```

Comment:
Instead of using `ulid` here, which is not guaranteed to be strictly ordered, we should use the transaction commit timestamp.

See https://pgpedia.info/p/pg_xact_commit_timestamp.html. This is the time the transaction commits, not the time that the transaction starts. We are currently using `NOW()` which returns the time that `NOW()` is called, which could return a timestamp before the time at which the data is actually committed, and thus it could lead to ordering inconsistencies.

Reply:
Even if i change `inserted_at` to use `pg_xact_commit_timestamp`, it will not fix the problem because every item writen in the transaction would share this value.
<img width="843" alt="image" src="https://github.com/openfga/openfga/assets/5374887/7a0a593c-58d0-4042-b9ad-445dc91d27dd">

Reply:
Gotcha 👍